libr/anal/sign.c

```diff
@@ -1113,21 +1113,51 @@ R_API bool r_sign_delete(RAnal *a, const char *name) {
 	return sdb_remove (a->sdb_zigns, k, 0);
 }
 
+static ut8 * build_combined_bytes(RSignBytes *bsig) {
+	if (!bsig || !bsig->bytes || !bsig->mask) {
+		return NULL;
+	}
+	ut8 *buf = (ut8 *)malloc(bsig->size);
+	if (!buf) {
+		return NULL;
+	}
+	int i;
+	for (i = 0; i < bsig->size; i++) {
+		buf[i] = bsig->bytes[i] & bsig->mask[i];
+	}
+	return buf;
+}
+
+static double cmp_bytesig_to_buff(RSignBytes *sig, ut8 *buf, int len) {
+	if (!sig || !buf || len < 0) {
```

Comment:
use `r_return_val_if_fail`. These checks actually should always be true, otherwise there is something very wrong and just ignoring it is not good because it hides the real problem (e.g. someone is passing wrong values to this function).

Reply:
Sorry, I thought, for some reason, `r_return_val_if_fail` was used for `R_API` functions. I will fix these.

Reply:
You can use it whenever you want to ensure there is no "programmatic error" (eg to check some conditions that you assume should always be true at some point of the program)